compiler/rustc_lint/src/levels.rs

```diff
@@ -283,7 +283,15 @@ impl<'s> LintLevelsBuilder<'s> {
         for (attr_index, attr) in attrs.iter().enumerate() {
             if attr.has_name(sym::automatically_derived) {
```

Comment:
The old implementation skipped the computation altogether for `automatically_derived` code. This was in part motivated by performance reasons. It could be relevant still, to close the performance gap.